pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -676,8 +677,18 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
 				if err != nil {
 					return err
 				}
-				var preRestoreObjects []runtime.Unstructured
-				if !nfs {
+				var preRestoreObjects, objects []runtime.Unstructured
+				// this check is equivalent to (if nfs && driverName == volume.PortworxDriverName)
+				if nfs {
+					if restore.Spec.ReplacePolicy == storkapi.ApplicationRestoreReplacePolicyRetain {
+						// Skip pv/pvc if replacepolicy is set to retain to avoid creating with empty object
+						backupVolInfos, existingRestoreVolInfos, err = a.skipVolumesFromRestoreList(restore, objects, driver, vInfos)
```

Comment:
pass nil in the place of objects. and there is no need of declaration line at 680 when both preRestoreObjects and objects are only used in else part of the code. They dont need to be declared again and again in the loop.